src/main/asciidoc/java/parsetools.adoc

```diff
@@ -47,4 +47,164 @@ RecordParser.newFixed(4, h -> {
 });
 ----
 
-For more details, check out the `link:../../apidocs/io/vertx/core/parsetools/RecordParser.html[RecordParser]` class.
\ No newline at end of file
+For more details, check out the `link:../../apidocs/io/vertx/core/parsetools/RecordParser.html[RecordParser]` class.
+
+== Json Parser
+
+You can easily parse JSON structures but that requires to provide the JSON string at once, but it
+may not be convenient when you need to parse very large structures.
+
+The non-blocking JSON parser is an event driven parser able to deal with very large structures.
+It transforms a sequence of input buffer to a sequence of JSON parse events.
+
+[source, java]
+----
+JsonParser parser = JsonParser.newParser();
+
+// Set handlers for various events
+parser.handler(event -> {
+  switch (event.type()) {
+    case START_OBJECT:
+      // Start an objet
+      break;
+    case END_OBJECT:
+      // End an objet
+      break;
+    case START_ARRAY:
+      // Start an array
+      break;
+    case END_ARRAY:
+      // End an array
+      break;
+    case VALUE:
+      // Handle a value
+      String field = event.field();
+      if (field != null) {
+        // In an object
+      } else {
+        // In an array or top level
+        if (event.isString()) {
+
+        } else {
+          // ...
+        }
+      }
+      break;
+  }
+});
+----
+
+The parser is non-blocking and emitted events are driven by the input buffers
+
+[source, java]
+----
+JsonParser parser = JsonParser.newParser();
+
+// start array event
+// start object event
+// "firstName":"Bob" event
+parser.handle(Buffer.buffer("[{\"firstName\":\"Bob\"},"));
```

Comment:
I think the end parenthesis shouldn't be there.

Reply:
it's valid java code

Reply:
Yes the Java code compiles. I was wondering if the parenthesis inside the JSON content would not trigger an end object event. Am I wrong?

Reply:
I don't see parenthesis in JSON content

Reply:
you mean curly brace ?

Reply:
Ha. Yeah, I meant curly brace. Sorry.

Reply:
the content was wrong, I think I fixed it now